lib/server.js

```diff
@@ -1366,6 +1369,22 @@ Server.prototype._finishReqResCycle = function _finishReqResCycle(
         var finalErr = err || res.err;
         req.emit('restifyDone', route, finalErr);
         self.emit('after', req, res, route, finalErr);
+    } else if (
+        res._handlersFinished === true &&
+        res.headersSent === false &&
+        !res.err
+    ) {
+        // if we reached the end of the handler chain and headers haven't been
+        // sent AND there isn't an existing res.err (e.g., req abort/close),
+        // it's possible it's a user error and a response was never written.
+        // send a 500.
+        res.send(
```

Comment:
Should we also do the stuff from the previous if block here? Decrement inflight requests, emitting after, etc?

It seems like we are asserting along this path that this is an error, why wouldn't we clean up?

Reply:
Calling `res.send` will actually loop back to this method again, It's probably more accurate to rename `finishReqResCycle` to `attemptFinishReqResCycle`, as we call it optimistically from various points in the code base and let this method handle the state reconciliation.